runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/HistoryTable/HistoryTable.tsx

```diff
@@ -102,6 +102,19 @@ const HistoryTable = forwardRef<any, HistoryTableProps>(({ botId, processId, sx,
         },
     }));
 
+    const handleRowExpand = (currRow) => {
+        if (currRow.subRows.length > 0 || currRow.isExpanded) return;
+        dispatch(processInstanceActions.getSubProcesses({ processInstanceId: currRow.original.id }))
+            .then(unwrapResult)
+            .catch(() => {
+                enqueueSnackbar(
+                    translate('History.Table.Error.SubProcessesNotFound'),
+                    { variant: 'error' },
+                );
+                dispatch(processInstanceActions.updateProcessInstance({ id: currRow.original.id, hasSubProcesses: false }));
+            });
+    };
+
```

Comment:
imo this whole logic can be in file `HistoryTable.columns.tsx` you have all information needed there just import some hooks. It's desired to move logic outside of the main components

Reply:
noted!
done